src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -88,15 +87,62 @@
 
 @postgres_dialect.segment(replace=True)
 class FunctionDefinitionGrammar(BaseSegment):
-    """This is the body of a `CREATE FUNCTION AS` statement."""
+    """This is the body of a `CREATE FUNCTION AS` statement.
+
+    Options supported as defined in https://www.postgresql.org/docs/9.1/sql-createfunction.html
+    """
 
     match_grammar = Sequence(
-        "AS",
-        OneOf(Ref("QuotedLiteralSegment"), Ref("DollarQuotedLiteralSegment")),
+        AnyNumberOf(
+            Sequence("TRANSFORM", "FOR", "TYPE", Ref("ParameterNameSegment")),
+            OneOf("WINDOW", "IMMUTABLE", "STABLE", "VOLATILE", "STRICT"),
+            Sequence(OneOf("NOT", optional=True), "LEAKPROOF"),
```

Comment:
I'm not 100% sure how to read tthe spec](https://www.postgresql.org/docs/9.5/sql-createfunction.html) but why is WINDOW on a separate line in that but not here? Even if it doesn't matter it might be good to align with the spec. Similar for `LEAKPROOF`, and leaving `STRICT` until the next line:

```suggestion
            Ref("WINDOW"),
            OneOf("IMMUTABLE", "STABLE", "VOLATILE", Sequence(Ref("NOT", optional=True), "LEAKPROOF"),),
```

Reply:
Okay, so the reason it's on a separate line in the Postgres spec, is because it's a separate option. The spec reads that you can have IMMUTABLE or VOLATILE or STABLE, but the only other option for WINDOW is nothing. This comment probably works as an answer to a few of them, but the way i've grouped things in my implementation is based on syntax structure, not their logical position in the function definition. So all options that are just one keyword are together, keyword followed by param are together, keyword - numeric are together, etc. I think we need a new logical operator (though there is a chance this may already exist and i cant find it). The operator should be like AsNumberOf, but limit the number of times each of the *args can appear. Then we could use it to list the options by logical structure, and it would be meaningful. A test case being, "WINDOW WINDOW" would pass in this current implementation, but shouldn't, but if we could pop each of the *args when its found, we could engineer that to fail.

Reply:
Actually you CAN have what you want here! `AnyNumberOf` takes an optional `max_times` argument so just add `max_times=1` to line 140 and you can avoid "WINDOW WINDOW"!!

Reply:
I've just tried it out and I think max_times is 'the number of times a thing matches' rather than 'the number of times each thing matches'

Reply:
Yes think you are right. So it's set to `1` for `OneOf` but unset for `AnyNumberOf`. Shame. Looks like we need a `no_dups` optional parameter for this `AnyNumberOf`. Anyway, let's ignore for now...